### ISSUE TITLE
dmzsm and pmzsm return NaN when x > J

### ISSUE THREAD

**Issue by piklprado** [piklprado/sads#69]
`dzsm` returns NaN for any abundance value larger than the community size J:

```r
> dmzsm(x = 100, J = 99, theta = 10)
[1] NaN
Warning messages:
1: In log(sn) : NaNs produzidos
2: In dmzsm(x = 100, J = 99, theta = 10) : NaNs produced
```
And so does `pzsm`:

```r
> pmzsm(q = 100, J = 99, theta = 10)
[1] NaN
Warning message:
In pmzsm(q = 100, J = 99, theta = 10) : NaNs produced
```
While the correct values to be returned are zero and one, respectively. 

As noted by jansen.vargas@gmail.com,  this error is catched by the validity check of class `octav`, that does not allow negative or NaN values in frequencies:

```r
> dados=c(1,1,1,1,1,2,2,3,3,5,6,8,8,8,9,12,13,13,17,38,38,47,56,154,389)
> dados.mzsm <- fitsad(dados, sad ="mzsm")
> octavpred(dados.mzsm)
Error in if (any(object[, 2] < 0) | any(object[, 3] < 0)) return("No negative values are allowed in the upper and Freq columns") : 
  valor ausente onde TRUE/FALSE necessário
Além disso: Warning message:
In (function (q, J, theta, lower.tail = TRUE, log.p = FALSE)  :
  NaNs produced
```

**Comment by piklprado** [comment]
Fixed with commit 14c306e. Please check.

**Comment by piklprado** [comment]
Fixed in PR#73
